### ISSUE TITLE
[gateway-gcs] Multipart Upload hangs

### ISSUE THREAD

**Issue by krisis** [minio/minio#4405]
Upload of a 81MiB file hangs. Following is the `mc --debug` output, we can observe that listIncompleteUploads is called repeatedly. This may be the possible root cause.
```
 0 B / 80.91 MB ┃░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░┃   0.00%mc: <DEBUG> GET /krisis/?max-uploads=1000&prefix=go1.6.2.linux-amd64.tar.gz&uploads= HTTP/1.1
Host: 192.168.1.10:9000
User-Agent: Minio (linux; amd64) minio-go/2.0.4 mc/2017-05-20T10:10:19Z
Authorization: AWS4-HMAC-SHA256 Credential=minio/20170523/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=**REDACTED**
X-Amz-Content-Sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
X-Amz-Date: 20170523T131908Z
Accept-Encoding: gzip

mc: <DEBUG> HTTP/1.1 200 OK
Transfer-Encoding: chunked
Accept-Ranges: bytes
Content-Type: application/xml
Date: Tue, 23 May 2017 13:19:30 GMT
Server: Minio/DEVELOPMENT.2017-05-23T11-56-08Z (linux; amd64)
Vary: Origin
X-Amz-Bucket-Region: 
X-Amz-Request-Id: 14C13EC5B18A5B27

mc: <DEBUG> Response Time:  21.919580353s

mc: <DEBUG> GET /krisis/?max-uploads=1000&prefix=go1.6.2.linux-amd64.tar.gz&uploads= HTTP/1.1
Host: 192.168.1.10:9000
User-Agent: Minio (linux; amd64) minio-go/2.0.4 mc/2017-05-20T10:10:19Z
Authorization: AWS4-HMAC-SHA256 Credential=minio/20170523/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=**REDACTED**
X-Amz-Content-Sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
X-Amz-Date: 20170523T131930Z
Accept-Encoding: gzip

mc: <DEBUG> HTTP/1.1 200 OK
Transfer-Encoding: chunked
Accept-Ranges: bytes
Content-Type: application/xml
Date: Tue, 23 May 2017 13:19:52 GMT
Server: Minio/DEVELOPMENT.2017-05-23T11-56-08Z (linux; amd64)
Vary: Origin
X-Amz-Bucket-Region: 
X-Amz-Request-Id: 14C13ECAE93F2AFA

mc: <DEBUG> Response Time:  22.380619955s

```
## Expected Behavior

Multipart Upload hangs.

## Current Behavior

Multipart Upload should complete successfully.

## Possible Solution

Not yet known.

## Steps to Reproduce (for bugs)

1. `mc cp /path/to/largefile mygcs/bucket`
2. `mc` hangs
3.
4.

## Your Environment

* Version used (`minio version`):
```
➜  minio git:(feature-gcs) ✗ minio version
Version: 2017-05-23T11:56:08Z
Release-Tag: DEVELOPMENT.2017-05-23T11-56-08Z
Commit-ID: 8c30dcddee6bae7b83519cee8983dfe881317c6b

```
* Environment name and version (e.g. nginx 1.9.1):
* Server type and version:
* Operating System and version (`uname -a`):
* Link to your project:

**Comment by harshavardhana** [comment]
Could be related to this https://github.com/minio/minio/issues/4380

**Comment by krishnasrinivas** [comment]
https://github.com/minio/minio/pull/4455 fixes this

**Comment by harshavardhana** [comment]
Fixed by #4455
